src/SignalR/clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -13,6 +13,31 @@ import { JsonHubProtocol } from "./JsonHubProtocol";
 import { NullLogger } from "./Loggers";
 import { Arg, ConsoleLogger } from "./Utils";
 
+// tslint:disable:object-literal-sort-keys
+const LogLevelNameMapping = {
+    trace: LogLevel.Trace,
+    debug: LogLevel.Debug,
+    info: LogLevel.Information,
+    information: LogLevel.Information,
+    warn: LogLevel.Warning,
+    warning: LogLevel.Warning,
+    error: LogLevel.Error,
+    critical: LogLevel.Critical,
+    none: LogLevel.None,
+};
+
+function parseLogLevel(name: keyof typeof LogLevelNameMapping): LogLevel {
+    // Case-insensitive matching via lower-casing
+    // Yes, I know case-folding is a complicated problem in Unicode, but we only support
+    // the ASCII strings defined in LogLevelNameMapping anyway, so it's fine -anurse.
+    const mapping = LogLevelNameMapping[name.toLowerCase()];
+    if (typeof mapping !== "undefined") {
+        return mapping;
+    } else {
+        throw new Error(`Unknown log level: ${name}`);
```

Comment:
Nit: Do we want to preserve the casing on the user input here?

Reply:
This does preserve that casing.